src/sql-parser/src/parser.rs

```diff
@@ -7720,9 +7720,24 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_show(&mut self) -> Result<ShowStatement<Raw>, ParserError> {
+        let redacted = self.parse_keyword(REDACTED);
         if self.parse_one_of_keywords(&[COLUMNS, FIELDS]).is_some() {
+            if redacted {
+                return parser_err!(
+                    self,
+                    self.peek_prev_pos(),
+                    "SHOW REDACTED with COLUMNS or FIELD is not supported"
+                );
+            }
```

Comment:
Instead of using a bunch of if-statements, what if we did something like:
```
let redacted = self.parse_keyword(REDACTED);
if !self.peek_keyword(CREATE) {
  // ...return an error
}
```

Reply:
Good idea, thank you!